tests/common/test_cgroupstelemetry.py

```diff
@@ -44,6 +47,49 @@ def generate_metric_list(lst):
             len(lst)]
 
 
+def consume_cpu_time():
+    waste = 0
+    for x in range(1, 200000):
+        waste += random.random()
+    return waste
+
+
+def consume_memory():
+    waste = []
+    for x in range(1, 3):
+        waste.append([random.random()] * 10000)
+        time.sleep(0.1)
+        waste *= 0
+    return waste
+
+
+def make_root_cgroup():
+    cpu_controller_id = CGroupsApi._get_controller_id('cpu')
```

Comment:
Try to write the tests using the public interface instead of private methods when possible. You could use the public interface to create a couple of test cgroups. Reaching into the private methods makes the tests more fragile (they may start failing or doing incorrect stuff if the internal implementation changes)